lang/rust/avro/src/types.rs

```diff
@@ -333,8 +333,15 @@ impl Value {
     /// See the [Avro specification](https://avro.apache.org/docs/current/spec.html)
     /// for the full set of rules of schema validation.
     pub fn validate(&self, schema: &Schema) -> bool {
+        let rs = ResolvedSchema::try_from(schema).expect("Schema didn't successfully parse");
+        self.validate_internal(schema, rs.get_names())
+    }
+
+    fn validate_internal(&self, schema: &Schema, names: &NamesRef) -> bool {
```

Comment:
Also what would you think about having this return a result? We should be able to reuse the resolved schema in the encode flow as well and propagate schema resolution errors up?

Reply:
I think this would be definitely an improvement!

Few more ideas:
1) the Java SDK recently gained support for `avropath` in the error messages ([AVRO-3375](https://issues.apache.org/jira/browse/AVRO-3375)
2) collect as much errors as possible while validating, not just one at a time [AVRO-3429](https://issues.apache.org/jira/browse/AVRO-3429)

Reply:
^ Yes, very much like this. Upping the validation errors would be huge.